src/react-native/12-storing-state/storing-state.md

```diff
@@ -0,0 +1,131 @@
+@page learn-react-native/storing-state Storing State in Navigation Parameters
+@parent learn-react-native 12
+@outline 3
+
+@description Maintain the React state with React Native Navigation Parameters
+
+@body
+
+## Overview
+
+In this section, you will:
+
+- Strongly type the navigation parameters of an application.
+- Maintain and pass the state using route params through navigation.
+
+## Objective 1: Intro to navigation parameters
+
+Now that we've successfully implemented React Navigation in our application, we can navigate between screens easily. The only remaining issue is that we lack away to pass information, or state, between screens. So, our new goal for this section is passing state between screens using navigation parameters.
+
+### Navigation Parameters
+
+As mentioned in the previous section, since our React Native application isn't navigated through URLs, we aren't able to pass the parameters through a URL. Instead, we'll be using the Stack we've already made.
+
+@sourceref ./StackRoute.tsx
+@highlight 4-23, 25
+
+Before we get into using `route` on each `Screen` of the `Navigator`, considering we're using TypeScript, we need to make an effort to make sure the properties for each component are properly typed. For this, we will create a type, `ShopStackParamList`.
+
+For each screen we will type the expected properties that will be passed along each route. The `Home` in this case doesn't expect any parameters to be passed to it, so we leave it undefined. The `UserProfile` and `Storefront` contain a few properties.
+
+Now, our `createStackNavigator` includes a type we've made `ShopStackParamList`. Because of this, now if we provide our screen components `Props` as route params, TypeScript will be able to able to identify what parameters are accessible from the components `route.params`.
+
+While the `route` is accessible from the `Navigator`, in order to utilize the parameters being passed by the it we can do so within each component instead.
```

Comment:
this paragraph is a bit confusing 😅 , if I understand the example correctly, we should rewrite it to something like:

> While the `route` is accessible from the `Navigator`, is also accessible from the component that is being navigated to through `props`

?

Reply:
Yeah, that's right, I added it and pushed it up.